kubernetes/client/kubernetes.go

```diff
@@ -0,0 +1,738 @@
+						},
+					},
+					Resources: resources,
+					SecurityContext: &apiv1.SecurityContext{
+						Privileged: boolPtr(true),
+						Capabilities: &apiv1.Capabilities{
+							Add: []apiv1.Capability{"SYS_ADMIN"},
+						},
+					},
+				},
+			},
+			InitContainers: []apiv1.Container{
+				{
+					Name:      "runner-init-container",
+					Image:     "ubuntu:latest",
+					Command:   []string{"/bin/tar", "-xf", "-"},
+					Stdin:     true,
+					StdinOnce: true,
+					VolumeMounts: []apiv1.VolumeMount{
+						{
+							Name:      "wr-temp",
+							MountPath: tempMountPath,
+						},
+					},
+				},
+			},
+		},
+	}
+	// Create pod
+	pod, err := p.podClient.Create(pod)
+	if err != nil {
+		p.Logger.Error("failed to create pod", "err", err)
+		return nil, err
+	}
+
+	/*
+	   No longer copying tarball to pod here, instead the deployment controller
+	   waits on the status of the InitContainer to be running, then runs
+	   CopyTar, found in pod.go
+
+	*/
+
+	return pod, err
+}
+
+// TearDown deletes the namespace and cluster role binding created for wr.
+func (p *Kubernetesp) TearDown(namespace string) error {
+	err := p.clientset.CoreV1().Namespaces().Delete(namespace, &metav1.DeleteOptions{})
+	if err != nil {
+		p.Logger.Error("deleting namespace", "err", err, "namespace", namespace)
+		return err
+	}
+
+	crbl, err := p.clientset.RbacV1().ClusterRoleBindings().List(metav1.ListOptions{
+		LabelSelector: "wr-" + namespace,
+	})
+	if err != nil {
+		p.Logger.Error("getting ClusterRoleBindings", "err", err)
+		return err
+	}
+	err = p.clientset.RbacV1().ClusterRoleBindings().Delete(crbl.Items[0].ObjectMeta.Name, &metav1.DeleteOptions{})
+	if err != nil {
+		p.Logger.Error("deleting ClusterRoleBinding", "ClusterRoleBinding", crbl.Items[0].ObjectMeta.Name, "err", err)
+		return err
+	}
+
+	return nil
+}
+
+// DestroyPod deletes the given pod, doesn't check it exists first.
+func (p *Kubernetesp) DestroyPod(podName string) error {
+	err := p.podClient.Delete(podName, &metav1.DeleteOptions{})
+	if err != nil {
+		p.Logger.Error("deleting pod", "err", err, "pod", podName)
+		return err
+	}
+	return nil
+}
+
+// CheckPod checks a given pod exists. If it does, return the status. Currently
+// not used anywhere func (p *Kubernetesp) CheckPod(podName string) (working
+// bool, err error) {
+//  pod, err := p.podClient.Get(podName, metav1.GetOptions{})
+//  if err != nil {
+//      return false, err
+//  }
+//  switch {
+//  case pod.Status.Phase == "Running":
+//      return true, nil
+//  case pod.Status.Phase == "Pending":
+//      return true, nil
+//  case pod.Status.Phase == "Succeeded":
+//      return false, nil
+//  case pod.Status.Phase == "Failed":
+//      return false, nil
+//  case pod.Status.Phase == "Unknown":
+//      return false, nil
+//  default:
+//      return false, nil
+//  }
+// }
+
+// NewConfigMap creates a new configMap Kubernetes 1.10 (Released Late March
+// 2018) provides a BinaryData field that could be used to replace the
```

Comment:
Punctuation...

Reply:
Argh, I've re-wrapped everything and now I've got a problem where I used to have new lines.